src/js/typography.js

```diff
@@ -0,0 +1,78 @@
+let FontFaceObserver = require('fontfaceobserver/fontfaceobserver.standalone.js');
+
+class Typography {
+
+	constructor (TypographyEl, opts) {
+		this.TypographyEl = TypographyEl;
+		this.opts = opts || { values: "default" };
+
+		this.fontLoadingPrefix = 'o-typography--loading-';
+		this.fontLoadedCookieName = 'o-typography-fonts-loaded';
+		this.fontConfigs = [
+			{
+				family: 'FinancierDisplayWeb',
+				weight: 'normal',
+				label: 'serifDisplay'
+			},
+			{
+				family: 'MetricWeb',
+				weight: 'normal',
+				label: 'sans'
+			},
+			{
+				family: 'MetricWeb',
+				weight: 600,
+				label: 'sansBold'
+			},
+			{
+				family: 'FinancierDisplayWeb',
+				weight: 700,
+				label: 'serifDisplayBold'
+			}
+		];
+
+		this.loadFonts();
+	}
+
+	removeLoadingClasses() {
+		this.fontConfigs.forEach((config) => {
+			this.TypographyEl.classList.remove(`${this.fontLoadingPrefix}${config.label}`);
+		});
+	}
+
+	loadFonts() {
+		if (new RegExp(`(^|\\s)${this.fontLoadedCookieName}=1(;|$)`).test(document.cookie)) {
```

Comment:
Just an FYI we've had issues occasionally with cookie length so we tend to use local storage instead